src/jsx/import/result/results.jsx

```diff
@@ -8,120 +8,122 @@
-                this.setState({ajax_fail_error: asynchError});
-                console.log('Error Thrown: ' + asynchError);
-            }
-        // boolean to hide ajax spinner
-            this.setState({display_spinner: false});
-        }.bind(this),
-      // pass ajax arguments
-        ajaxArguments);
-    },
-    render: function() {
-      // local variables
+                    // enumerate and store response
+                    this.setState(Object.assign({}, results));
+                }
+                // boolean to hide ajax spinner
+                this.setState({ display_spinner: false });
+            },
+            // asynchronous callback: ajax 'fail' promise
+            (asynchStatus, asynchError) => {
+                if (asynchStatus) {
+                    this.setState({ ajax_fail_status: asynchStatus });
+                    console.log(`Error Status: ${asynchStatus}`);
+                }
+                if (asynchError) {
+                    this.setState({ ajax_fail_error: asynchError });
+                    console.log(`Error Thrown: ${asynchError}`);
+                }
+                // boolean to hide ajax spinner
+                this.setState({ display_spinner: false });
+            },
+            // pass ajax arguments
+            ajaxArguments,
+        );
+    }
+    render() {
+        // local variables
         const status = this.state.status;
         const titles = this.state.titles;
-        var spinner = this.state.display_spinner ? <Spinner /> : null;
+        const spinner = this.state.display_spinner ? <Spinner /> : null;
 
-      // polyfill 'entries'
+        // polyfill 'entries'
         if (!Object.entries) {
             entries.shim();
         }
 
-      // generate result
+        // generate result
         if (status == 0 && titles && titles.length > 0) {
-            var resultList = <ul className='result-list'>{
+            var resultList = (<ul className='result-list'>{
                 titles.map((title) => {
                     if (title.length == 3) {
-                        return <NavLink
-                            to={'/session/current-result?nid=' + title[0]}
-                            key={'link-' + title[0]}
+                        return (<NavLink
+                            to={`/session/current-result?nid=${title[0]}`}
+                            key={`link-${title[0]}`}
                         >
-                            <li key={'title-' + title[0]}>
+                            <li key={`title-${title[0]}`}>
                                 {title[0]}: {title[1]}
                             </li>
-                        </NavLink>
+                        </NavLink>);
                     }
                 })
-            }</ul>;
+            }
+                              </ul>);
         } else {
-            var resultList = <div className='result-list'>
+            var resultList = (<div className='result-list'>
```

Comment:
Being picky, does it make sense to have the content follow after the parentheses, on a newline?